### DISCUSSION TITLE
Upgrade From Xamarin to AvaloniaUI

### DISCUSSION THREAD

**Discussion by raselldev** [AvaloniaUI/Avalonia#17580] [Q&A]
I want migrate Xamarin.iOS and Xamarin.Android project to another framework. Is good if i migrate to AvaloniaUI or migrate to MAUI? But in MAUI still many bugs. I confuse to choose framework for migration Xamarin project.

**Comment by timunie** [comment]
Asking this here will lead to use Avalonia 🤣

Joke aside, it depends on your needs. If you want platform controls, Avalonia is not for you. If you want your UI being almost the same on each platform you found your solution. 

My tip is to mock up a prototype and see if it works for you

**Reply by raselldev** [reply]
Our management say we need to stay in .NET environment, developer dont want fight with many bugs in MAUI. FYI our management only know about Xamarin.

**Reply by timunie** [reply]
Avalonia is .NET so you are welcome. We also have paid support for you available if needed. Reach out via web form in case you are interested.

**Reply by thevortexcloud** [reply]
Do keep in mind while you can make your own mobile apps in Avalonia, out of the box what you get is fairly minimal. You either need to make a lot of things yourself or use third party libraries such as:

https://github.com/AvaloniaInside/Shell
https://github.com/AvaloniaCommunity/Material.Avalonia
